### PR TITLE
feat: Skip name duplicate checking when explaining

### PR DESCRIPTION
**Pull request by yinfredyue**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## Summary

This PR addresses #5189. Previously, when generating a plan for creating a new index/sink/table/mv, an error might be thrown for name duplicates. This PR lifts the duplicate-checking code out of plan-generation functions, such that when generating plans for explaining, no name duplicate error is thrown.

## Details

- Make `check_relation_name_duplicated` a pure check and returns `Result<()>`;
- Make `gen_create_xxx_plan`'s call `get_schema_by_name`, `get_database_by_name` directly.
- Make actual handlers (`handle_xxx`) call `check_relation_name_duplicated`

## Manual testing
```
dev=> create table t (v int);
CREATE_TABLE
dev=> explain create table t (v int);
                                 QUERY PLAN
----------------------------------------------------------------------------
 StreamMaterialize { columns: [v, _row_id(hidden)], pk_columns: [_row_id] }
   StreamExchange { dist: HashShard(_row_id) }
     StreamSource { source: "t", columns: [v, _row_id] }
(3 rows)
dev=> create materialized view mv as select v from t order by v limit 10;
CREATE_MATERIALIZED_VIEW
dev=> explain create materialized view mv as select v from t order by v limit 10;
                                                 QUERY PLAN
-------------------------------------------------------------------------------------------------------------
 StreamMaterialize { columns: [v, t._row_id(hidden)], pk_columns: [t._row_id], order_descs: [v, t._row_id] }
   StreamProject { exprs: [t.v, t._row_id] }
     StreamTopN { order: "[t.v ASC]", limit: 10, offset: 0 }
       StreamExchange { dist: Single }
         StreamGroupTopN { order: "[t.v ASC]", limit: 10, offset: 0, group_key: [2] }
           StreamProject { exprs: [t.v, t._row_id, Vnode(t._row_id)] }
             StreamTableScan { table: t, columns: [v, _row_id] }
(7 rows)
dev=> create index idx_test on t (v);
CREATE_INDEX
dev=> explain create index idx_test on t (v);
                                                 QUERY PLAN
-------------------------------------------------------------------------------------------------------------
 StreamMaterialize { columns: [v, t._row_id(hidden)], pk_columns: [t._row_id], order_descs: [v, t._row_id] }
   StreamExchange { dist: HashShard(t.v) }
     StreamTableScan { table: t, columns: [v, _row_id] }
(3 rows)
```

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

* SQL commands, functions, and operators

@skyzh @xxchan @st1page

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
Could you please add some sqllogictests? Might look like this:

```
statement ok
create ...

# explain doesn't check name is duplicated
statement ok
explain create ...

# create will do
statement error
create ...
```

File: src/frontend/src/handler/create_mv.rs [hunk 1]
```diff
@@ -117,7 +108,17 @@ pub async fn handle_create_mv(
     let session = context.session_ctx.clone();
 
     let (table, graph) = {
-        let (plan, table) = gen_create_mv_plan(&session, context.into(), query, name, false)?;
+        {
+            let catalog_reader = session.env().catalog_reader().read_guard();
```
Comment (xxchan) [inline src/frontend/src/handler/create_mv.rs:112]: An additional `read_guard` is introduced, which makes me still prefer `get` and `check` at the same place. How do you think?

Comment (xxchan) [reply src/frontend/src/handler/create_mv.rs:112]: > Also, does (database, schema) uniquely identify a relation? The function returns Result<(DatabaseId, SchemaId)>, but you said we should name it get_relation_name.

Oops, I finally understand what you mean previously..

I think we can lookup `db` and `schema` outside and pass them to `check_relation_name_duplicated`.

**Comment by yinfredyue** [comment]
> Could you please add some sqllogictests? Might look like this:
> 
> ```
> statement ok
> create ...
> 
> # explain doesn't check name is duplicated
> statement ok
> explain create ...
> 
> # create will do
> statement error
> create ...
> ```

Sure. But I am not familiar with how tests are organized in RisingWave. Should I add it to `src/frontend/planner_test`, or `src/tests`, or `src/frontend/handler/create_xx.rs`?

**Comment by xxchan** [comment]
> Sure. But I am not familiar with how tests are organized in RisingWave. Should I add it to `src/frontend/planner_test`, or `src/tests`, or `src/frontend/handler/create_xx.rs`?
> 
> Can you point me to some existing test(s) that I can use as a reference?

There's a `e2e_test/batch/explain.slt`, so you may have a `e2e_test/batch/explain_existing_table.slt`

You can read [Developer guide](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#end-to-end-tests) to know more about tests

**Review by skyzh — APPROVED** [review]
LGTM from my side, shall we add some sqllogictest about this case?

**Comment by yinfredyue** [comment]
SLT added! @skyzh @xxchan

**Comment by yinfredyue** [comment]
`misc-check` fails for some reason. It seems like it cannot pull `shellcheck` from the registry? Weird...

**Comment by skyzh** [comment]
It happens sometimes... You can push an empty commit by `git commit -m "trigger CI" --allow-empty` to rerun.

File: src/frontend/src/handler/create_mv.rs (hunk 1 above)
Comment (yinfredyue) [reply src/frontend/src/handler/create_mv.rs:112]: Seems like @skyzh  is ok with this. I will mark this as resolved.

**Review by xxchan — APPROVED** [review]

**Pull request merged**